cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -98,6 +98,17 @@ Result<std::shared_ptr<SchemaManifest>> GetSchemaManifest(
   return manifest;
 }
 
+bool isNan(const Scalar& value) {
+  if (value.type->Equals(*float32())) {
```

Comment:
Doing a full type equality check is unnecessarily slow here
```suggestion
  if (value.type->id() == Type::FLOAT) {
```

Reply:
Made the change, thanks!